File: app/src/main/java/io/github/wulkanowy/ui/login/LoginContract.java
package io.github.wulkanowy.ui.login;

import io.github.wulkanowy.di.annotations.PerActivity;
import io.github.wulkanowy.ui.base.BaseContract;

public interface LoginContract {
    interface View extends BaseContract.View {

        void requestPasswordViewFocus();

        void requestEmailViewFocus();

        void setPasswordError(String message);

        void setEmailError(String message);

        void resetViewErrors();

        void hideSoftKeyboard();

    }

    @PerActivity
    interface Presenter extends BaseContract.Presenter<View> {

        void attemptLogin(String email, String password, String symbol);
    }
}

File: app/src/main/java/io/github/wulkanowy/ui/login/LoginContract.kt
package io.github.wulkanowy.ui.login

import io.github.wulkanowy.ui.base.BaseContract

interface LoginContract {
    interface View : BaseContract.View {

        fun setErrorEmailRequired()

        fun setErrorPassRequired()

        fun setErrorEmailInvalid()

        fun setErrorPassInvalid()

        fun setErrorPassIncorrect()

        fun resetViewErrors()

        fun setStepOneLoginProgress()

        fun setStepTwoLoginProgress()

        fun openMainActivity()

        fun showLoginProgress(show: Boolean)

        fun showSoftKeyboard()

        fun hideSoftKeyboard()

        fun showActionBar(show: Boolean)

        fun onSyncFailed()

    }

    interface Presenter : BaseContract.Presenter<View> {

        fun attemptLogin(email: String, password: String)
    }
}
